site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java

```diff
@@ -13182,159 +13126,46 @@ private void setSelectedParticipantRoles(SessionState state) {
 
 	} // setSelectedParticipantRol3es
 
+	@Getter @AllArgsConstructor
 	public class MyIcon {
-		protected String m_name = null;
-
-		protected String m_url = null;
-
-		protected String m_skin = null;
-
-		public MyIcon(String name, String url, String skin) {
-			m_name = name;
-			m_url = url;
-			m_skin = skin;
-		}
-
-		public String getName() {
-			return m_name;
-		}
-
-		public String getUrl() {
-			return m_url;
-		}
-
-		public String getSkin() {
-			return m_skin;
-		}
+		private String name = null;
+		private String url = null;
+		private String skin = null;
 	}
 
 	// a utility class for working with ToolConfigurations and ToolRegistrations
 	// %%% convert featureList from IdAndText to Tool so getFeatures item.id =
 	// chosen-feature.id is a direct mapping of data
+	@Getter @EqualsAndHashCode
```

Comment:
Can we even have multiple instances of SiteAction, with different state? Bloody weird. If we want to sit on this until after 12 is cut, cool. On the other hand if we want to actually steadily improve SiteAction then maybe getting it into the QA process for 12 might be a good idea?